engine/access/access_test.go

```diff
@@ -107,6 +131,15 @@ func (suite *Suite) SetupTest() {
 
 	suite.chainID = flow.Testnet
 	suite.metrics = metrics.NewNoopCollector()
+
+	suite.finalizationDistributor = pubsub.NewFinalizationDistributor()
+	suite.finalizedHeaderCache, _ = synceng.NewFinalizedHeaderCache(suite.log, suite.state, suite.finalizationDistributor)
```

Comment:
let's check the error to avoid hard to debug test failures
```suggestion
	suite.finalizedHeaderCache, err := synceng.NewFinalizedHeaderCache(suite.log, suite.state, suite.finalizationDistributor)
	require.NoError(suite.T(), err)
```